platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,217 @@
+- Try to remove imports from `platform/monitoring` and instead import the relevant code in outside of the forms library and pass them in as props or config options.
+  - As an example, in the [`FormSignInModal`](https://github.com/department-of-veterans-affairs/vets-website/blob/1cc955b8d4f6b9f93f4553fdd4afa9878c75564f/src/platform/forms/save-in-progress/FormSignInModal.jsx#L12) component, remove the `platform/monitoring/record-event` import and instead put that function call inside of the function that gets passed as a prop. `FormSignInModal` is only used from `platform/site-wide`
+- Remove forms library dependence on platform re-implementations of 3rd party functions, like lodash's `get` and `set`.
+  - Some of these examples will be easy to avoid the lodash functions entirely, but for others we should switch to the lodash functions instead of the platform versions, with a longer term goal being to get away from lodash altogether
+- Remove imports for `src/platform/utilities/environment` and instead rely on global `const`s from webpack's DefinePlugin for environment-related `const`s (i.e. an api url) or branching
+
+
+#### Final touches
+
+Once the forms library is actually a package, app code can be updated to import from it directly instead of relying on the aliases.
+
+- Each app team will update their code to use forms library JS & SCSS from the package instead of by direct path
+- Configure the `no-restricted-imports` ESLint rule to [restrict imports of `src/platform/packages` directly](https://eslint.org/docs/rules/no-restricted-imports)
+- Remove the `.babelrc` forms-library alias & custom SASS importer
+
+### Code Location
+
+The forms library package will live in `src/platform/packages/forms-library`.
+Since it will be a Node package, it will be added to the `package.json` file and apps will import it as a regular Node package instead of by its direct path.
+
+### Testing Plan
+
+We have ESLint rules in place which can verify that the right JS is imported as it relates to the forms library.
+Additionally, many of the end-to-end tests will fail if the forms library is not imported properly, so they will help with awareness.
+
+We don't have any automation in place to validate that the right SASS files are imported (other than webpack failing to resolve the file), so testing that forms styling is working properly will be a manual process using the browser dev tools.
```

Comment:
How might we automate this?

Reply:
Visual regression testing :grin:

Reply:
We could set up local VRT, actually. That's not too painful, I think. It's automated VRT that's the beast.